crates/hir/src/symbols.rs

```diff
@@ -242,45 +242,84 @@ impl<'a> SymbolCollector<'a> {
         <<L as Lookup>::Data as HasSource>::Value: HasName,
     {
         self.push_file_symbol(|s| {
+            let mut symbols = Vec::new();
```

Comment:
Instead of allocating a vec everytime, let's just remove the `push_file_symbol` abstraction and push manually into the `SymbolCollector::symbols` vec for performance.